Business Social Network/AppModules/Search/views/searchApp.js

```diff
@@ -1,8 +1,51 @@
 var search = angular.module("searchApp", []);
 
-search.controller('SearchBoxController', function($scope, $http){
+search.controller('SearchBoxController', function($scope, $location){
 
-  var userSearch;
+  //show all categories by default
+  $scope.selected = "All Categories";
+
+  //get all categories to display in dropdown menu
+  //dummy data:
+  $scope.categories = ["Clothes", "Electronics", "Books", "Bateekh"];
+  var selectedCategory = "All Categories";
+  var userSearch = ' ';
+
+  $scope.update = function(){
+
+    userSearch = $scope.search_term;
+
+    //update URL if necessary
+    if(userSearch.length > 0){
+      $location.path('/search/' + selectedCategory + '/' + userSearch);
+    }
+
+    else{
+      $location.path('/');
+    }
+
+  }
+
+  $scope.updateCategory = function(category){
+
+    $scope.selected = category;
+    selectedCategory = category;
+
+  //update URL if necessary
+    if(userSearch.length > 0){
+      $location.path('/search/' + selectedCategory + '/' + userSearch);
+    }
+
+    else{
+      $location.path('/');
```

Comment:
@Swailem95 
if no search results are available angular should route to the previous route not the root route